proxy/src/coco.rs

```diff
@@ -1,649 +1,358 @@
-pub fn blob(
-    paths: &Paths,
-    id: &str,
-    maybe_revision: Option<String>,
-    maybe_path: Option<String>,
-) -> Result<Blob, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let meta = project::Project::show(paths, &project_id)?;
-
-    let mut browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
-
-    let path = maybe_path.clone().unwrap_or_default();
-    let revision = maybe_revision.unwrap_or(meta.default_branch);
-
-    // Best effort to guess the revision.
-    browser.revspec(&revision)?;
-
-    let root = browser.get_directory()?;
-
-    let mut p = surf::file_system::Path::from_str(&path)?;
-
-    let file = root.find_file(&p).ok_or_else(|| {
-        surf::file_system::error::Error::Path(surf::file_system::error::Path::Empty)
-    })?;
-
-    let mut commit_path = surf::file_system::Path::root();
-    commit_path.append(&mut p);
-
-    let last_commit = browser.last_commit(&commit_path)?.map(|c| Commit::from(&c));
-    let (_rest, last) = p.split_last();
-    let content = match std::str::from_utf8(&file.contents) {
-        Ok(content) => BlobContent::Ascii(content.to_string()),
-        Err(_) => BlobContent::Binary,
-    };
-
-    Ok(Blob {
-        content,
-        info: Info {
-            name: last.to_string(),
-            object_type: ObjectType::Blob,
-            last_commit,
-        },
-        path: maybe_path.unwrap_or(last.to_string()),
-    })
-}
-
-/// Given a project id to a repo returns the list of branches.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn branches(paths: &Paths, id: &str) -> Result<Vec<Branch>, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
-
-    let mut branches = browser
-        .list_branches(None)
-        .expect("Getting branches failed")
-        .into_iter()
-        .map(|b| Branch(b.name.name().to_string()))
-        .collect::<Vec<Branch>>();
-
-    branches.sort();
-
-    Ok(branches)
-}
 
-/// Given a path to a repo returns the list of branches.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the repository doesn't exist.
-pub fn local_branches(repo_path: &str) -> Result<Vec<Branch>, error::Error> {
-    let repo = surf::git::Repository::new(repo_path)?;
-    let browser = surf::git::Browser::new(repo)?;
-    let mut branches = browser
-        .list_branches(None)?
-        .into_iter()
-        .map(|b| Branch(b.name.name().to_string()))
-        .collect::<Vec<Branch>>();
-
-    branches.sort();
-
-    Ok(branches)
-}
+    /// Fetch a browser for the `project_urn` we supplied to this function.
```

Comment:
```suggestion
    /// Fetch a repository for the `project_urn` we supplied to this function.
```

Says `browser` here but actually return the `repository`.